src/ChunkDataCallback.h

```diff
@@ -101,3 +101,37 @@ class cChunkDataSeparateCollector :
 
 
 
+/** A simple implementation of the cChunkDataCallback interface that just copies the cChunkData */
+class cChunkDataCopyCollector :
+	public cChunkDataCallback
+{
+public:
+	struct MemCallbacks:
+		cAllocationPool<cChunkData::sChunkSection>::cStarvationCallbacks
+	{
+		virtual void OnStartUsingReserve() override {}
+		virtual void OnEndUsingReserve() override {}
+		virtual void OnOutOfReserve() override {}
+	};
+
+	cChunkDataCopyCollector():
+		m_Pool(cpp14::make_unique<MemCallbacks>()),
+		m_Data(m_Pool)
+	{
+	}
+
+
+	cListAllocationPool<cChunkData::sChunkSection, cChunkData::NumSections> m_Pool;  // Keep 1 chunk worth of reserve
```

Comment:
I don't think this is a good idea, as we are only using this pool for 1 chunk data so there no point having a pool. Again, in hindsight I think the pool adds no value, and we need to put more thought into how we handle large buffer.